lib/response/headers_builder.ex

```diff
@@ -0,0 +1,30 @@
+defmodule HTTPServer.Response.HeadersBuilder do
+  alias HTTPServer.Response.Headers
+
+  def build, do: %Headers{}
```

Comment:
Build is super generic.  What if you wanted to use this "functional builder pattern" for other things.  Maybe be a little more specific here will help. Or alternatively, do not import but alias instead when you call?

example call:
```exlixir
 alias HTTPServer.Response.HeadersBuilder
...

headers =
      HeadersBuilder.build()
      |> content_length(body)
      |> content_type()
      |> host(req.headers)
      |> allow(methods)
```

Reply:
I think if I were to alias, then every function called to build the headers would have to include `HeadersBuild.` in front of it. When I originally wrote it that way it felt like it was getting very wordy. But I agree that build is unclear, so to improve clarity, I'm down to change it to:
```
headers =
      HeadersBuilder.build()
      |> HeadersBuilder.content_length(body)
      |> HeadersBuilder.content_type()
      |> HeadersBuilder.host(req.headers)
      |> HeadersBuilder.allow(methods)
```
or maybe I could increase specificity for the `build()` function? And change it to something like:
```
headers =
      build_headers()
      |> content_length(body)
      |> content_type()
      |> host(req.headers)
      |> allow(methods)
```

Reply:
yeah, I didn't think about having to rename then all.  I like the second suggestion in this case.

Reply:
though, does it matter that they are Response headers? (I do think you will be building Request headers ever, but just pointing it out)

Reply:
As yet another option?

```Elixir
headers =
      %Headers{}
      |> content_length(body)
      |> content_type()
      |> host(req.headers)
      |> allow(methods)
```

Reply:
Actually your `%Headers{}` suggestion is WAY simpler 😅 I'll go with that!